### PR TITLE
Adding privacy-preserving Concrete ML links (#9598)

### PR DESCRIPTION
**Pull request by bcm-at-zama**
Adding links to privacy-preserving Concrete ML library.

refs #9598 

CC @trivialfis 
CC @rongou

### CODE REVIEW

**Comment by bcm-at-zama** [comment]
Hello, my first PR in XGBoost so sorry if I missed something. At least, I have checked that `make html` works fine.

File: doc/prediction.rst [hunk 1]
```diff
@@ -146,3 +146,48 @@ instance we might accidentally call ``clf.set_params()`` inside a predict functi
 
     with ThreadPoolExecutor(max_workers=10) as e:
         e.submit(predict_fn, ...)
+
+*****************************
+Privacy-Preserving Prediction
+*****************************
+
+Concrete ML is a third-party open-source library developed by `Zama`_ that proposes gradient
```
Comment (rongou) [inline doc/prediction.rst:154]: I'd put the link to `Concrete ML` here instead of the last sentence.

File: doc/prediction.rst [hunk 2]
```diff
@@ -146,3 +146,48 @@ instance we might accidentally call ``clf.set_params()`` inside a predict functi
...
+boosting classes similar to ours, but predicting directly over encrypted data, thanks to
+so-called Fully Homomorphic Encryption. A simple `Concrete ML`_ example would be as follows:
```
Comment (rongou) [inline doc/prediction.rst:156]: No need for "so-called".

Probably just say "A simple example would be as follows".

File: doc/prediction.rst (hunk 1 above)
Comment (bcm-at-zama) [reply doc/prediction.rst]: You're right, thx

**Comment by bcm-at-zama** [comment]
Have taken into account the feedback from @rongou

**Review by rongou — APPROVED** [review]

**Comment by bcm-at-zama** [comment]
Thanks @rongou . Sorry, newbie here, I don't know if I have to do anything to make the builds green

**Comment by trivialfis** [comment]
Triggered the CI.

**Comment by bcm-at-zama** [comment]
Still some failing tests in the CI

**Comment by bcm-at-zama** [comment]
```
* branch                refs/pull/9604/head -> FETCH_HEAD
--
  | # FETCH_HEAD is now `c41e7aa7a0087a09ce2d5af280729add8dd713b1`
  | $ git checkout -f 00cdc9c44c1f8ff408a4569d53b4558df0cbf2be
  | fatal: reference is not a tree: 00cdc9c44c1f8ff408a4569d53b4558df0cbf2be
  | ⚠️ Warning: Checkout failed! exit status 128 (Attempt 3/3)
  | 🚨 Error: exit status 128
```

and 00cdc9c44c1f8ff408a4569d53b4558df0cbf2be is the new commit in this branch. I don't know what's wrong, maybe someone can guide me a bit

**Comment by trivialfis** [comment]
Those are fine. We don't need heavy tests for document changes.

**Review by trivialfis — APPROVED** [review]

**Comment by trivialfis** [comment]
Thank you for the doc!

**Comment by bcm-at-zama** [comment]
Thanks a lot @trivialfis

**Pull request merged**